TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyDecodedHttpRequestConvertHandler.java

```diff
@@ -0,0 +1,312 @@
+                        + "all content handlers are being loaded."
+                );
+                // make a fresh message and its headers
+                requestPipelineOrchestrator.addJsonParsingHandlers(
+                    ctx,
+                    transformer,
+                    getAuthTransformerAsStreamingTransformer(authTransformer)
+                );
+                ctx.fireChannelRead(handleAuthHeaders(httpJsonMessage, authTransformer));
+            }
+        } else if (msg instanceof HttpContent) {
+            ctx.fireChannelRead(msg);
+        } else {
+            assert false
+                : "Only HttpRequest and HttpContent should come through here as per RequestPipelineOrchestrator";
+            // In case message comes through, pass downstream
+            super.channelRead(ctx, msg);
+        }
+    }
+
+    private static HttpJsonRequestWithFaultingPayload transform(
+        IJsonTransformer transformer,
+        HttpJsonRequestWithFaultingPayload httpJsonMessage
+    ) {
+        var originalHttpJsonMessage = httpJsonMessage;
+        var beforeContentHeaders = getEncodingHeaders(httpJsonMessage);
+
+        ByteBuf innerPayloadByteBuf = null;
+        ByteBuf protectedByteBuf = null;
+        try {
+            innerPayloadByteBuf = (ByteBuf) httpJsonMessage.payload().get(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY);
+            if (innerPayloadByteBuf != null) {
+                protectedByteBuf = Unpooled.unreleasableBuffer(innerPayloadByteBuf);
+                httpJsonMessage.payload().put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY,
+                    protectedByteBuf);
+            }
+        } catch (PayloadNotLoadedException e) {
+            // Skip byteBuf protection if payload not loaded
+        }
+
+        var returnedObject = transformer.transformJson(httpJsonMessage);
+
+        if (returnedObject != httpJsonMessage) {
+            httpJsonMessage = new HttpJsonRequestWithFaultingPayload(returnedObject);
+        }
+
+        if (innerPayloadByteBuf != null) {
+            // replace protected byteBuf if was hidden and still there
+            var transformedByteBuf = httpJsonMessage.payload().get(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY);
+            assert protectedByteBuf != null : "Expected protectedByteBuf to be defined if innerPayloadByteBuf is";
+            if (protectedByteBuf.equals(transformedByteBuf)) {
+                httpJsonMessage.payload().put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY, innerPayloadByteBuf);
+            }
+        }
+
+        var afterContentHeaders = getEncodingHeaders(httpJsonMessage);
+
+        if (!Objects.deepEquals(beforeContentHeaders, afterContentHeaders)) {
+            // Ensure payload was loaded during transformations if modified content headers
+            originalHttpJsonMessage.payload().forEach((key, val) -> {});
```

Comment:
This fixed a bug in the existing request transforms. If using the canned addGzip transform, it would be run and detected as a "headersOnlyTransform" so this detects when to certain content headers necessitate resetting the pipeline with the content handlers.